src/NuGet.Core/NuGet.Configuration/Utility/SettingsUtility.cs

```diff
@@ -289,7 +289,7 @@ public static IReadOnlyList<string> GetFallbackPackageFolders(ISettings settings
                 paths[i] = Path.GetFullPath(paths[i]);
             }
 
-            return paths;
+            return paths.AsReadOnly();
```

Comment:
This becomes tricky because we have public APIs that we wouldn't want to duplicate unless we had to. 

If we remove our good API design hats, return a `List` here and using `AsList` above is likely the most performant implementation, List implements both `IList` and `IReadOnlyList`.

Reply:
Fwiw, I'm ok with keeping this change, as given that we're already using ToList, there's no perf change here.

Reply:
If there is no perf change do we still need to make this change?

Reply:
It's still a "correctness" change.  Without this change, a caller could do `((List<T>)result)[0] = newValue`. With this change that would fail. I don't know whether or not we want to do that though. It is off-topic in a PR titled "remove ToList() usage".

Reply:
Given that `GetFallbackPackageFolders` method currently returns `IReadOnlyList<string>`, I am okay with this change unless CI fails because of this.